mcs/class/corlib/System.Reflection/FieldInfo.cs

```diff
@@ -222,6 +223,55 @@ internal object[] GetPseudoCustomAttributes ()
 			return attrs;
 		}
 
+		internal CustomAttributeData[] GetPseudoCustomAttributesData ()
+		{
+			int count = 0;
+
+			if (IsNotSerialized)
+				count++;
+
+			if (DeclaringType.IsExplicitLayout)
+				count++;
+
+			MarshalAsAttribute marshalAs = get_marshal_info ();
+			if (marshalAs != null)
+				count++;
+
+			if (count == 0)
+				return null;
+			CustomAttributeData[] attrsData = new CustomAttributeData [count];
+			count = 0;
+
+			if (IsNotSerialized)
+				attrsData [count++] = CustomAttributeData.Create (
```

Comment:
Add ctor overload with `ConstructorInfo ctorInfo` only instead of passing extra 2 empty arguments at many places